app/delivery/send_to_providers.py

```diff
@@ -201,35 +204,49 @@ def load_provider(provider_id: str) -> ProviderDetails:
+    """
 
-    if is_feature_enabled(FeatureFlag.PROVIDER_STRATEGIES_ENABLED):
-        provider = provider_service.get_provider(notification)
-        return clients.get_client_by_name_and_type(provider.identifier, notification.notification_type)
+    try:
+        if is_feature_enabled(FeatureFlag.PROVIDER_STRATEGIES_ENABLED):
+            provider = provider_service.get_provider(notification)
+            return clients.get_client_by_name_and_type(provider.identifier, notification.notification_type)
 
-    if is_feature_enabled(FeatureFlag.TEMPLATE_SERVICE_PROVIDERS_ENABLED):
-        provider_id = get_provider_id(notification)
+        if is_feature_enabled(FeatureFlag.TEMPLATE_SERVICE_PROVIDERS_ENABLED):
+            provider_id = get_provider_id(notification)
+
+            if provider_id:
+                return clients.get_client_by_name_and_type(
+                    load_provider(provider_id).identifier,
+                    notification.notification_type
+                )
 
-        if provider_id:
-            return clients.get_client_by_name_and_type(
-                load_provider(provider_id).identifier,
-                notification.notification_type
+        # This is a list of ProviderDetails instances sorted by their "priority" attribute.
+        active_providers_in_order = [
+            p for p in get_provider_details_by_notification_type(
+                notification.notification_type,
+                notification.international
             )
+            if should_use_provider(p)
+        ]
 
-    # This is a list of ProviderDetails instances sorted by their "priority" attribute.
-    active_providers_in_order = [
-        p for p in get_provider_details_by_notification_type(notification.notification_type, notification.international)
-        if should_use_provider(p)
-    ]
+        if not active_providers_in_order:
+            current_app.logger.error(
+                f"{notification.notification_type} {notification.id} failed as no active providers"
+            )
+            raise RuntimeError(f"No active {notification.notification_type} providers")
 
-    if not active_providers_in_order:
-        current_app.logger.error(
-            f"{notification.notification_type} {notification.id} failed as no active providers"
+        # This returns an instance of one of the classes defined in app/clients/.
+        return clients.get_client_by_name_and_type(
+            active_providers_in_order[0].identifier,
+            notification.notification_type
         )
-        raise Exception(f"No active {notification.notification_type} providers")
-
-    # This returns an instance of one of the classes defined in app/clients/.
-    return clients.get_client_by_name_and_type(active_providers_in_order[0].identifier, notification.notification_type)
+    except ValueError as e:
+        current_app.logger.error("Couldn't retrieve a client for the given provider.")
+        current_app.logger.exception(e)
```

Comment:
Update logger
```suggestion
        current_app.logger.exception("%s", e)
```

Reply:
Adding "%s" here seems like an extra step. Is it the proper way to handle it?

Reply:
It will ensure that the exception is printed to the log as a string, so this is not extra, it's proper! :)